packages/avatars/src/styled/utility.ts

```diff
@@ -16,40 +16,35 @@ const [xxs, xs, s, m, l] = SIZE;
 export const TRANSITION_DURATION = 0.25;
 
 export interface IStyledStatusIndicatorProps extends ThemeProps<DefaultTheme> {
-  readonly size?: IAvatarProps['size'];
-  readonly type?: IAvatarProps['status'] | 'active';
+  readonly $size?: IAvatarProps['size'];
+  readonly $type?: IAvatarProps['status'] | 'active';
 }
 
 export function getStatusColor(
-  type?: IStyledStatusIndicatorProps['type'],
-  theme?: IStyledStatusIndicatorProps['theme']
+  theme: IStyledStatusIndicatorProps['theme'],
+  type?: IStyledStatusIndicatorProps['$type']
 ): string {
-  switch (type) {
-    case 'active':
-      return getColorV8('crimson', 400, theme)!;
-    case 'available':
-      return getColorV8('mint', 400, theme)!;
-    case 'away':
-      return getColorV8('orange', 400, theme)!;
-    case 'transfers':
-      return getColorV8('azure', 400, theme)!;
-    case 'offline':
-      return getColorV8('grey', 500, theme)!;
-    default:
-      return 'transparent';
-  }
```

Comment:
Def a [nit] and related to the micro-optimization comment above, the switch statement is slightly more performant then calculating all colors then picking the right one.

Reply:
Let me know if I captured the intent of this optimization in my latest commit. I think I did but after reading your comment I'm not entirely sure.

Reply:
Yeah, I was more thinking the map would live in a one-time constant. And the function call would be more along the lines of https://github.com/zendeskgarden/react-components/blob/816c29f24a9353b2abc320c0c1019db83fa1b018/packages/chrome/src/styled/utils.ts#L24-L25

Reply:
The one time constant would def work. 

What I wanted to emphasize is that when using a map 5 values are calculated every time `getStatusColor` is called but only one color is returned. With a switch statement, however, there's one calculation for one returned color. It's more efficient.

Reply:
Can you be more specific @jzempel ? These functions feel very different to me.

Reply:
This was my take on it: 403bf68d9c3fd10076501bf31c9501cc99c84de2

Thoughts?

Reply:
I dig it! 💯

Reply:
Hey @geotrev maybe something like this?

```ts
const STATUS_COLOR_OPTIONS: Record<
  NonNullable<IStyledStatusIndicatorProps['$type']>,
  Omit<ColorParameters, 'theme'>
> = {
  active: { hue: 'crimson', light: { shade: 700 }, dark: { shade: 600 } },
  available: { hue: 'mint', light: { shade: 500 }, dark: { shade: 400 } },
  away: { hue: 'orange', light: { shade: 500 }, dark: { shade: 400 } },
  transfers: { hue: 'azure', light: { shade: 500 }, dark: { shade: 400 } },
  offline: { hue: 'grey', light: { shade: 500 }, dark: { shade: 400 } }
};

export function getStatusColor(
  theme: IStyledStatusIndicatorProps['theme'],
  type?: IStyledStatusIndicatorProps['$type']
): string {
  return type ? getColor({ ...STATUS_COLOR_OPTIONS[type], theme }) || 'transparent' : 'transparent';
}
```

Reply:
Haha, looks like we arrived at a similar result :wink:

Reply:
Two peas 😎

Reply:
I like retaining the `'transparent'` fallback in case the color enum lookup returns `undefined`, so I added that in. :)